src/front/wgsl/mod.rs

```diff
@@ -1809,6 +1809,64 @@ impl Parser {
 
                 Some(crate::Statement::Loop { body, continuing })
             }
+            "for" => {
+                let mut block = Vec::new();
+
+                lexer.expect(Token::Paren('('))?;
+                let initialization = self.parse_statement(lexer, context.reborrow(), false)?;
+                let condition = self.parse_general_expression(lexer, context.as_expression())?;
+                lexer.expect(Token::Separator(';'))?;
+                let mut continuing = None;
+                if let Token::Word(ident) = lexer.peek() {
```

Comment:
let's make a clear comment on why this is needed versus `parse_statement`, so that nobody gets confused :)